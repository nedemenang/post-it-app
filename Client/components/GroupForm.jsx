import React, {Component} from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom';
import '../public/style.css';
import $ from '../public/jquery.js';
import AppActions from '../actions/AppActions';



class GroupForm extends Component {

  /**
   * Creates group
   * @param {object} event event object
  * @return {void} return void
  */
submit(event){
  event.preventDefault();


  let today = new Date();
  let dd = today.getDate();
  let mm = today.getMonth()+1; //January is 0!
  let yyyy = today.getFullYear();

  if(dd<10) {
      dd = '0'+dd
  } 

  if(mm<10) {
      mm = '0'+mm
  } 

  today = mm + '/' + dd + '/' + yyyy;

  const groupname = this.refs.group.value.trim();
  const dateCreated = today;

  const user = localStorage.getItem('user');

 if(this.refs.group.value !== '')
  {
    let group = {
      groupname : groupname,
      datecreated: dateCreated,
      createdBy: JSON.parse(user).email, 
      createdByDisplayName: JSON.parse(user).displayName,
      createdByProfilePic: JSON.parse(user).profilePic,
      createdByUserId: JSON.parse(user).id
    }
    // console.log('Creating Group');
    AppActions.createGroup(group); 
    this.refs.group.value = '';
    $('.group-form').slideToggle();
  }
}

  constructor(props){
    super(props);
    this.state= {};
    this.submit = this.submit.bind(this);
  }
  render(){
    return(
      <div className="bottomMargin">
         <form onSubmit={this.submit} className= "group-form">
              <input type="text" className="form-control" ref="group" placeholder="Press enter to submit" />
              <p>{this.props.errors}</p>
         </form>
      </div>
    );
  }
}

export default GroupForm;