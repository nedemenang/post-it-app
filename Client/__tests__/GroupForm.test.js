import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import appAction from '../actions/AppActions';
import GroupForm from '../components/GroupForm.jsx';

describe('the group form component', () => {
  let mountedComponent;
  jest.mock('../dispatcher/AppDispatcher');

  const str = location.href;
  const str2 = str.replace('#', ' ');

  const loggedInUser = {
    displayName: 'nnamisco',
    email: 'nedemenang@gmail.com',
    id: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    isAuthenticated: true,
    phoneNo: null,
    profilePic: `${str2}static/files/blank-profile-pic.png`
  };

  const groupFormItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<GroupForm loggedInUser = {loggedInUser} key = {1}/>);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = groupFormItem();
    expect(component).toBeDefined();
  });

  it('should match snapshot test', () => {
    const component = groupFormItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it('should recieve props', () => {
    expect(Object.keys(groupFormItem().props()).length).toBeGreaterThan(0);
  });

  it('should call function on submit of form', () => {
    const component = shallow(<GroupForm loggedInUser = {loggedInUser}/>);
    const preventDefault = jest.fn();
    component.find('form').simulate('submit', { preventDefault });
    expect(preventDefault).toBeCalled();
  });

  it('should update state on input field change', () => {
    const input = groupFormItem().find('input').first();
    input.simulate('change', { target: {
      value: 'New Group'
    } });
    expect(groupFormItem().state().groupname).toBe('New Group');
  });
});
