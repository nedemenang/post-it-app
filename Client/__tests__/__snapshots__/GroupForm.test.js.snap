// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the group form component should match snapshot test 1`] = `
<div
  className="bottomMargin"
>
  <form
    className="group-form"
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Press enter to submit"
      type="text"
      value=""
    />
    <p />
  </form>
</div>
`;
